doc/introduction-to-xkb.md

```diff
@@ -0,0 +1,200 @@
+# Introduction to XKB {#xkb-intro}
+
+__XKB__ stands for “X Keyboard Extension”. It may refer to either:
+
+- a [protocol](@ref xkb-the-protocol)
+- a [keyboard layout configuration](@ref xkb-the-config)
+- a [text format](@ref xkb-the-text-format)
+
+## XKB the protocol {#xkb-the-protocol}
+
+A __protocol__ for the [X Windows System], that extends the core protocol.
+
+_xkbcommon’s_ API is somehow derived from this API, but has been
+substantially reworked to expose fewer internal details to clients.
+
+_xkbcommon_ does not depend on a particular windows system; for instance
+it is used by the [Wayland] protocol.
+
+_xkbcommon_ provides the <code>[xkbcommon-x11]</code> module to interface
+a client with a X server. Relevant links:
+
+- [The X Window System Protocol][X Protocol]
+- [The X Keyboard Extension: Protocol Specification][XKB Protocol]
+- [xkbcommon-x11]
+
+
+## XKB the keyboard layout configuration {#xkb-the-config}
+
+In order to use [the protocol](@ref xkb-the-protocol), one must first load a
+[keyboard layout]. The layout usually comes from the OS _layout database_,
```

Comment:
"keyboard layout" might be confusing since it's only a component of the keymap. So maybe use the term "keymap" instead?

Reply:
Yes, this is tricky. I think it is more common for user to use “layout”, but here we have to be precise. I made some adjustements.